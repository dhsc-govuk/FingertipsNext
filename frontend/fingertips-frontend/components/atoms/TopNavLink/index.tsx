import { GovukColours } from '@/lib/styleHelpers/colours';
import NextLink from 'next/link';
import styled from 'styled-components';

export const TopNavLink = styled(NextLink)({
  'color': GovukColours.White,
  'textDecoration': 'none',
  'textDecorationSkipInk': 'none',
  'borderBottom': '1px solid transparent',
  'fontWeight': 700,
  'lineHeight': 1,
  ':hover': {
    borderBottomColor: GovukColours.White,
  },
  'display': 'inline-block',
  ':focus': {
    'color': GovukColours.Black,
    'backgroundColor': GovukColours.Yellow,
    'outline': `3px solid ${GovukColours.Yellow}`,
    ':hover': {
      borderBottomColor: GovukColours.Black,
    },
  },
});
