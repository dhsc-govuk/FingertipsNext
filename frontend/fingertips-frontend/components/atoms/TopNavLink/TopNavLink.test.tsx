import { TopNavLink } from '@/components/atoms/TopNavLink/index';
import { render, screen } from '@testing-library/react';

describe('TopNavLink', () => {
  it('should render with expected content', () => {
    const linkText = 'Link text';
    const linkUrl = 'https://www.example.com';
    render(<TopNavLink href={linkUrl}>{linkText}</TopNavLink>);

    const link = screen.getByRole('link');
    expect(link).toBeInTheDocument();
    expect(link).toHaveTextContent(linkText);
    expect(link).toHaveAttribute('href', linkUrl);
  });

  it('should match the snapshot', () => {
    const topNavLink = render(<TopNavLink href={'#'}>Link text</TopNavLink>);

    expect(topNavLink.asFragment()).toMatchSnapshot();
  });
});
