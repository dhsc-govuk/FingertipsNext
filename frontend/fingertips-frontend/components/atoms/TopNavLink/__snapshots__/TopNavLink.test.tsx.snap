// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavLink should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-text-decoration-skip-ink: none;
  text-decoration-skip-ink: none;
  border-bottom: 1px solid transparent;
  font-weight: 700;
  line-height: 1;
  display: inline-block;
}

.c0:hover {
  border-bottom-color: #FFFFFF;
}

.c0:focus {
  color: #0B0C0C;
  background-color: #FFDD00;
  outline: 3px solid #FFDD00;
}

.c0:focus:hover {
  border-bottom-color: #0B0C0C;
}

<a
    class="c0"
    href="#"
  >
    Link text
  </a>
</DocumentFragment>
`;
